src/codegen/compile.ml

```diff
@@ -8994,14 +9123,30 @@ let rec compile_lexp (env : E.t) ae lexp =
     store_ptr ^^
     get_field ^^
     compile_add_const ptr_unskew ^^
-    E.call_import env "rts" "write_barrier"
+    E.call_import env "rts" "post_write_barrier"
+  | IdxLE (e1, e2), Flags.Incremental when potential_pointer (Arr.element_type env e1.note.Note.typ) ->
+    let (set_array, get_array) = new_local env "array" in
```

Comment:
I'd add a todo to the new cases here to try to optimized them in future. I worry that the use of locals will hurt us and wonder if it might be better to push the value on the stack in the preamble and then call a function to do the postamble. But probably not worth investigating now - just leave a TODO, possibily linking a GH issue.

Reply:
Update: I removed the allocation increments and postponed the time to the scheduled GC increments. Thanks @ulan for this suggestion. This makes many things easier: We do not have to care about temporarily resolved pointers during assignments and other operations between the usual regularly scheduled GC calls. As a result, also the barriers are not cheap again, without needing local variables.